src/xml/XMLTag.cpp

```diff
@@ -140,10 +140,10 @@ Eigen::VectorXd XMLTag::getEigenVectorXdAttributeValue(const std::string &name,
   // std::map<std::string, XMLAttribute<utils::DynVector> >::const_iterator iter;
   auto iter = _eigenVectorXdAttributes.find(name);
   PRECICE_ASSERT(iter != _eigenVectorXdAttributes.end());
-  PRECICE_CHECK(iter->second.getValue().size() >= dimensions,
-                "Vector attribute \"" << name << "\" of tag <" << getFullName()
-                                      << "> has less dimensions than required (" << iter->second.getValue().size()
-                                      << " instead of " << dimensions << ")!");
+  const auto size = iter->second.getValue().size();
+  PRECICE_CHECK(size == dimensions,
+                "Vector attribute \"" << name << "\" of tag <" << getFullName() << "> is "
+                                      << size << "D, which does not match the dimension of the " << dimensions << "D SolverInterface.");
```

Comment:
Suggestion for consistency with the XML tags:
```suggestion
                                      << size << "D, which does not match the dimension of the " << dimensions << "D solver-interface.");
```